ingestion/src/metadata/utils/engines.py

```diff
@@ -38,7 +38,7 @@ def get_engine(workflow_source: WorkflowSource, verbose: bool = False) -> Engine
         options = {}
     engine = create_engine(
         get_connection_url(service_connection_config),
-        **options.dict(),
+        **options,
```

Comment:
@ayush-shah @OnkarVO7, in this [PR](https://github.com/open-metadata/OpenMetadata/pull/4027/files#diff-7ba89ab2f4820db88ff3d78709d291699775198b5e7d8ca751144469504cd41fR41) we were adding this `dict()` to `options`. This breaks other ingestions such as with MSSQL.

We have this issue https://github.com/open-metadata/OpenMetadata/issues/4032 to make the engine creation a bit smarter. For now, I am reverting this change, but we should fix the Bigquery ingestion and check how we are passing the parameters.

Reply:
Here I'd do two things to get this right and easy for us and users:
1. We know the parameters we need to prepare the BQ connection, let's move them to the JSON Schema
2. Prepare a singledispatch for the engine creation as we do with the URL. One default process where we unpack the `options`, and specific configs for those services where we have required params, such as BQ or the `http_path` in databricks
WDYT?

Reply:
Yes i think both points are ok we can do that